### ISSUE TITLE
[FEATURE] No clear Authentication mechanism documented for VertexAI Chat Models

### ISSUE THREAD

**Issue by Sachu1992** [langchain4j/langchain4j#461]
**Is your feature request related to a problem? Please describe.**
It has been a frustrating experience to build a VertexAI langchain integration in Java using Service account. This is because the documentation on Authentication is poor. 
The developer has no idea how to include the .json file while creating the VertexAiChatModel object. 

**Describe the solution you'd like**
A clear documentation on how to authenticate the VertexAiChatModels using .json files. 
In future, it will be good to accept the API Key for VertexAiChatModel similar to OpenAiChatModel instead of configuring environment variables in a roundabout way. 

**Describe alternatives you've considered**
I even added the json path to the System Environment variables with the key "GOOGLE_APPLICATION_CREDENTIALS". 
The Google documentation on Authentication is equally bad.
Similar Chat Model classes for OpenAi/Azure Open AI have no issue since they accept API key as a parameter

**Comment by langchain4j** [comment]
@Sachu1992 thank you for reporting and so sorry for frustration!
I have [added](https://github.com/langchain4j/langchain4j/pull/463/files) javadoc regarding authentication, could you please review? Thank you!

cc @glaforge FYI

**Comment by tomhillgreyridge** [comment]
I would agree that the setup of this is extremely confusing if you are coming at this fresh.  I've tried following the instructions in the Javadoc which worked for the simpler AI APIs but not for Gemini!

I managed to get an application running using vertex ai (text-bison) by creating a service account, creating a key, downloading it and storing on my local machine and setting the GOOGLE_APPLICATION_CREDENTIALS as an environment variable.  Calling text-bison then worked perfectly.

I then tried to change to use gemini-pro by changing to vertex-ai-gemini and the same project, same location now gives

`UNAUTHENTICATED: Request had invalid authentication credentials. Expected OAuth 2 access token, login cookie or other valid authentication credential.`

Is there anything different that needs to be done for Gemini?

**Comment by langchain4j** [comment]
@glaforge could you please take a look if you have some time?

**Comment by glaforge** [comment]
I have a project where I'm using both the PaLM and Gemini model, and I didn't see that problem.
Usually, when running locally, I don't use the `GOOGLE_APPLICATION_CREDENTIALS` approach with service account JSON keys. I just run `gcloud auth application-default login` and the underlying library recognizes I'm authenticated.
Then when I deploy my app to Google Cloud (for example on Cloud Run), it also figures out the environment it's running in, and is authenticated transparently as well.

Back when I wasn't using LangChain4j and I was interacting with the REST API directly, I was using the 
library and was creating a bearer token that way:
```
       var credentials = GoogleCredentials.getApplicationDefault()
                .createScoped('https://www.googleapis.com/auth/cloud-platform');
        credentials.refreshIfExpired();
        var token = credentials.getAccessToken().getTokenValue()
 ```

Not sure this helps. 
But I agree that authentication on GCP can be confusing :-(

**Comment by langchain4j** [comment]
@glaforge I faced a similar authentication issue when I tried to run integration tests with Gemini on github actions. Regular VertexAI works fine with `google-github-actions/auth@v2`: https://github.com/langchain4j/langchain4j/blob/main/.github/workflows/release-4-rest.yaml
But Gemini fails to auth, so I had to disable it for the moment.

**Comment by honeyshine75** [comment]
gemini keys has supported, will langchain4j support this?

**Comment by langchain4j** [comment]
@honeyshine75 can you elaborate please?

**Comment by honeyshine75** [comment]
> @honeyshine75 can you elaborate please?

What I mean is that we can use key to make post requests and thus use gemini
Like this：
curl --location 'https://generativelanguage.googleapis.com/v1/models/gemini-pro:generateContent?key=xxx-xxx' \
--header 'Content-Type: application/json' \
--data '{
    "generationConfig": {
        "maxOutputTokens": 2048,
        "temperature": 0.3,
        "topK": 10.0,
        "topP": 0.8
    },
    "contents": [
        {
            "parts": [
                {
                    "text": "hello"
                }
            ],
            "role": "user"
        }
    ]
}'

**Comment by glaforge** [comment]
The Vertex AI Gemini model in LangChain4j is using the Gemini LLM from Google Cloud, it's not the Gemini flavor offered by Google Labs. There are two flavors of Gemini. 
The LangChain4j currently only support the one in Google Cloud.
That said, maybe we could support both flavors at some point.

**Comment by kappamax** [comment]
I think I understand the problem people are having here, GOOGLE_APPLICATION_CREDENTIALS environment variable is deep in the vertex libraries and is expected to be a file path, not the actual json contents of the file.

So people who are using Vertex on other cloud providers and wanting to do it in a smart way without committing the json to their repo are confused how to get the credentials loaded from something like AWS Secrets Manager, and this works both on local as well as something like Beanstalk, ECS or AppRunner.

Here's a work around for that.

# Step 1
Set the following environment variables
GOOGLE_APPLICATION_CREDENTIALS = /tmp/<whatever_file_name.json>
GOOGLE_CREDENTIALS = arn for the secrets manager json for the credentials.

# Step 2
In your application bootstrap, eg. for Spring boot in your `EnvironmentPostProcessor`

Just read the secret, write it to a credentials file that you delet on exist.

```
        val credentialsFile = File(
            environment.getProperty("GOOGLE_APPLICATION_CREDENTIALS", "/tmp/credentials.json")
        )
        credentialsFile.deleteOnExit()
        credentialsFile.writeText(
            environment.getProperty("GOOGLE_CREDENTIALS", "")
        )
        credentialsFile.setReadOnly()
```

I still don't love this solution, as I prefer not to have the credential in a file, but I assume your cloud environment is properly locked down, I'd prefer if we could just read the secret into memory from secrets manager, but hopefully this helps someone who is stuck on this and trying to figure out how to make it work in a Cloud env.
